pyvo/utils/prototype.py

```diff
@@ -0,0 +1,191 @@
+import inspect
+import warnings
+from dataclasses import dataclass
+from functools import wraps
+from typing import Dict, Iterable
+
+from pyvo.dal.exceptions import PyvoUserWarning
+
+__all__ = ['features', 'prototype_feature', 'activate_features', 'Feature', 'PrototypeWarning', 'PrototypeError']
+
+features: Dict[str, "Feature"] = {
```

Comment:
Admittedly not very familiar with type hints but why `"Feature"` and not `Feature`?

Reply:
Without the quotes it would require `Feature` to be defined already. I like the code to be arranged in such a way that the most important information is at the top, then things follow in order of importance. The list of features seems to be more relevant than the nitty gritty details of the `Feature` class, so I put it first, which renders the `Feature` declaration unreachable from `features`, so we need a [Forward Reference](https://peps.python.org/pep-0484/#forward-references).

Reply:
That makes perfect sense. Thanks